File: aula-01/js/Ex21.js
/**
 * Algoritmo que certifica se o CPF é válido através do Método de Horner
 * @param {Number} d CPF a ser verificado
 * @throws {RangeError} caso d tenha mais ou menos de 11 dígitos
 * @returns {boolean} true caso seja verdadeiro, false caso contrário
 */
module.exports = CPF2;
function CPF2(d) {
    if (d.length != 11) {
        throw new RangeError("O CPF não pode conter mais ou menos que 11 dígitos");
    }
    let c = 8;
    let p = d[9];
    let s = d[9];
    while (c >= 1) {
        p = p + d[c];
        s = s + p;
        c = c - 1;
    }
    let j = (s % 11) % 10;
    let k = ((s - p + d[10]) % 11) % 10;
    return (j == d[10] && k == d[11]);
}

File: aula-01/java/Ex21.java
public class Ex21 {

    public static boolean CPF2(String d) {

        if (d.length() < 11 || d.length() > 11) {
            throw new IllegalArgumentException("O CPF deve conter 11 dígitos");
        }

        int c = 7;
        int p = Character.getNumericValue(d.charAt(8));

        // FIXME por que chamar novamente estes métodos???
        int s = Character.getNumericValue(d.charAt(8));

        // TODO Character.getNumericValue espalhado no código. Reúna em método.

        while (c >= 0) {
            p = p + Character.getNumericValue(d.charAt(c));
            s = s + p;
            c = c - 1;
        }

        int j = (s % 11) % 10;
        int k = ((s - p + 9 * Character.getNumericValue(d.charAt(9))) % 11) % 10;

        return (j == Character.getNumericValue(d.charAt(9)) && k == Character.getNumericValue(d.charAt(10)));

    }

}
